task_sdk/src/airflow/sdk/definitions/dag.py

```diff
@@ -769,10 +766,12 @@ def partial_subset(
         memo = {id(self.task_dict): None, id(self.task_group): None}
         dag = copy.deepcopy(self, memo)  # type: ignore
 
-        if isinstance(task_ids_or_regex, (str, Pattern)):
-            matched_tasks = [t for t in self.tasks if re2.findall(task_ids_or_regex, t.task_id)]
+        if isinstance(task_ids, str):
+            matched_tasks = [t for t in self.tasks if str(task_ids) in t.task_id]
```

Comment:
`task_ids` is already a str in this code path.

```suggestion
            matched_tasks = [t for t in self.tasks if task_ids in t.task_id]
```

Reply:
Yeap, removed